synapse/config/saml2_config.py

```diff
@@ -86,42 +90,64 @@ def read_config(self, config, **kwargs):
         )
         user_mapping_provider_config = user_mapping_provider_dict.get("config", {})
 
-        if user_mapping_provider_module == default_mapping_provider:
-            # Handle deprecated options in default module config
+        # Retrieve an instance of the module's class
+        # Pass the config dictionary to the module for processing
+        (
+            self.saml2_user_mapping_provider_class,
+            self.saml2_user_mapping_provider_config,
```

Comment:
the naming `saml2_user_mapping_provider_config` vs `user_mapping_provider_config` is confusing. Indeed you seem to have confused yourself in this PR. Can you come up with a better name for at least one of them?